src/app/components/shared/datetime-formats/datetime/datetime/datetime.component.spec.ts

```diff
@@ -0,0 +1,217 @@
+import { Spectator, createComponentFactory } from "@ngneat/spectator";
+import { SharedModule } from "@shared/shared.module";
+import { MockBawApiModule } from "@baw-api/baw-apiMock.module";
+import { DateTime, IANAZone, Settings, SystemZone } from "luxon";
+import { assertTooltip } from "@test/helpers/html";
+import { DatetimeComponent } from "./datetime.component";
+
+// I have created this interface for TypeScript LSP typing and auto completion
+// it should not be used outside of this file
+interface TestCase {
+  name: string;
+  value: Date | DateTime;
+  expectedText: string;
+  expectedTooltip: string;
+  expectedDateTimeAttribute: string;
+  date?: boolean;
+  time?: boolean;
+}
+
```

Comment:
If you make a function:

```
function testCase(name, value, expectedText, expectedTooltip, expectedAttribute, date, time) {
  return { name:, value:, expectedText:, expectedTooltip:, expectedAttribute:, date:, time: };
}
```

You could massive reduce the boiler plate in the following test case definitions

Reply:
Agreed, but I'd argue that it's harder to read

eg.

```js
{
    name: "DateTime with attribute time",
    value: DateTime.fromISO("2020-01-01T12:10:11.123Z"),
    expectedText: "2020-01-01",
    expectedTooltip: "2020-01-01 20:10:11.123 +08:00 Australia/Perth",
    expectedDateTimeAttribute: "2020-01-01T20:10:11.123+08:00",
    time: true,
},
```

would become:

```js
testCase(
	"DateTime with attribute time",
	DateTime.fromISO("2020-01-01T12:10:11.123Z"),
	"20:10:11.123",
	"2020-01-01 20:10:11.123 +08:00 Australia/Perth",
	"2020-01-01T20:10:11.123+08:00",
	false,
	true
)
```

It'd remove the need for us to specify keys in our test cases, but I don't see this as a win for readability

Reply:
You're not wrong. The trade-off is compactness making it easier to compare test cases.

Reply:
Your choice

Reply:
A dev preview of an implementation using the proposed factory method

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/69b27bc9-22fa-4562-a09e-3589ab99ea76)